addon/components/sl-grid.js

```diff
@@ -0,0 +1,660 @@
+
+        if ( renderedName ) {
+            var registry = this.get( 'container._registry' ),
+                root = renderedName.replace( '.', '/' ) + '/',
+                detailFooterPath = root + 'detail-footer',
+                detailHeaderPath = root + 'detail-header',
+                detailPath = root + 'detail',
+                filterPath = root + 'filter',
+                footerPath = root + 'footer';
+
+            if (
+                !this.get( 'detailFooterPath' ) &&
+                registry.resolve( 'template:' + detailFooterPath )
+            ) {
+                this.set( 'detailFooterPath', detailFooterPath );
+            }
+
+            if (
+                !this.get( 'detailHeaderPath' ) &&
+                registry.resolve( 'template:' + detailHeaderPath )
+            ) {
+                this.set( 'detailHeaderPath', detailHeaderPath );
+            }
+
+            if (
+                !this.get( 'detailPath' ) &&
+                registry.resolve( 'template:' + detailPath )
+            ) {
+                this.set( 'detailPath', detailPath );
+            }
+
+            if (
+                !this.get( 'filterPath' ) &&
+                registry.resolve( 'template:' + filterPath )
+            ) {
+                this.set( 'filterPath', filterPath );
+            }
+
+            if (
+                !this.get( 'footerPath' ) &&
+                registry.resolve( 'template:' + footerPath )
+            ) {
+                this.set( 'footerPath', footerPath );
+            }
+        }
+    }),
+
+    /**
+     * Whether to show the pagination in the list-pane footer
+     *
+     * @function
+     * @returns {Boolean}
+     */
+    showPagination: Ember.computed( 'continuous', 'totalPages', function() {
+        var totalPages = this.get( 'totalPages' );
+
+        return !this.get( 'continuous' ) && totalPages && totalPages > 1;
+    }),
+
+    /**
+     * The currently sorted column definition
+     *
+     * @function
+     * @returns {?Object} The definition for the currently sorted column
+     */
+    sortedColumn: Ember.computed( 'columns', 'sortedColumnTitle', function() {
+        var columns = this.get( 'columns' ),
+            sortedColumnTitle = this.get( 'sortedColumnTitle' );
+
+        if ( sortedColumnTitle ) {
+            for ( let i = 0; i < columns.length; i++ ) {
+                if ( Ember.get( columns[ i ], 'title' ) === sortedColumnTitle ) {
+                    return columns[ i ];
+                }
+            }
+        }
+    }),
+
+    /**
+     * The total number of pages of bound content, based on pageSize
+     *
+     * @function
+     * @returns {Number}
+     */
+    totalPages: Ember.computed(
+        'continuous', 'pageSize', 'totalCount',
+        function() {
+            if ( !this.get( 'continuous' ) ) {
+                return Math.ceil(
+                    this.get( 'totalCount' ) / this.get( 'pageSize' )
```

Comment:
How well are these trusted to be a numbers?  I haven't seen any init() logic that is forcing them to be, yet we are performing mathematical operations on them here.